### DISCUSSION TITLE
read file in static site

### DISCUSSION THREAD

**Discussion by mrkito** [flet-dev/flet#3146] [Q&A]
**Description**

**Code example to reproduce the issue:**

```python
from typing import Dict

import flet
from flet import (
    Column,
    ElevatedButton,
    FilePicker,
    FilePickerResultEvent,
    FilePickerUploadEvent,
    FilePickerUploadFile,
    Page,
    ProgressRing,
    Ref,
    Row,
    Text,
    icons,
)
import os

os.environ["FLET_SECRET_KEY"] = "3"


def main(page: Page):
    prog_bars: Dict[str, ProgressRing] = {}
    files = Ref[Column]()
    upload_button = Ref[ElevatedButton]()

    def file_picker_result(e: FilePickerResultEvent):
        upload_button.current.disabled = True if e.files is None else False
        prog_bars.clear()
        files.current.controls.clear()
        if e.files is not None:
            for f in e.files:
                prog = ProgressRing(value=0, bgcolor="#eeeeee", width=20, height=20)
                prog_bars[f.name] = prog
                files.current.controls.append(Row([prog, Text(f.name)]))
        page.update()

    def on_upload_progress(e: FilePickerUploadEvent):
        prog_bars[e.file_name].value = e.progress
        prog_bars[e.file_name].update()

    file_picker = FilePicker(on_result=file_picker_result, on_upload=on_upload_progress)

    def upload_files(e):
        uf = []
        if file_picker.result is not None and file_picker.result.files is not None:
            for f in file_picker.result.files:
                file_picker.result.files.save()
                uf.append(

                    FilePickerUploadFile(
                        f.name,
                        upload_url=page.get_upload_url(f.name, 600),
                    )
                )
            file_picker.upload(uf)

    # hide dialog in a overlay
    page.overlay.append(file_picker)

    page.add(
        ElevatedButton(
            "Select files...",
            icon=icons.FOLDER_OPEN,
            on_click=lambda _: file_picker.pick_files(allow_multiple=True),
        ),
        Column(ref=files),
        ElevatedButton(
            "Upload",
            ref=upload_button,
            icon=icons.UPLOAD,
            on_click=upload_files,
            disabled=True,
        ),
    )


flet.app(target=main, upload_dir="uploads", view=flet.WEB_BROWSER)

```

flet publish upload_example.py  

python3 -m http.server 55545 --directory dist 

**Describe the results you received:**
<img width="1134" alt="Screenshot_2024-04-22_at_15 34 09" src="https://github.com/flet-dev/flet/assets/29797481/6b82f05d-c03a-4bd9-b809-26f26e29c795">

**Describe the results you expected:**

i expect  get data from this file and just send to server

in linux,webapp  all is  ok, but in static site did not work

**Additional information you deem important (e.g. issue happens only occasionally):**

**Flet version (`pip show flet`)**:

```
Version: 0.22.0

```

**Operating system**: WEB

**Comment by FeodorFitsner** [comment]
So, you want to process the file on a client (browser) without uploading to a server?

**Reply by mrkito** [reply]
Yes

i wand choose file and just send to server without reading, but i cant see file in static web for send this file

this don't work in static web

**Reply by FeodorFitsner** [reply]
That's a good topic, we could return to it later.

**Reply by mrkito** [reply]
How we can read or send uploaded file in static site?

**Reply by mrkito** [reply]
@FeodorFitsner When will you want fix?

**Reply by mrkito** [reply]
@FeodorFitsner Hi) Will you fix upload file on static web?

**Reply by mpacheco62** [reply]
Hi, I'm also interested in reading and processing files directly on the client side. Furthermore, this data would not be sent to any server; I just need to read, process, and display the results.

**Comment by fangfufu** [comment]
Folks, I have a reply to this question: https://github.com/fangfufu/AmplifyP/blob/dev/docs/flet_static_file_handling.md

Demo: https://fangfufu.github.io/AmplifyP/

You can try loading this file: https://github.com/fangfufu/AmplifyP/blob/dev/amplify_gui_state.yaml and saving it (download it) again.

**Comment by ndonkoHenri** [comment]
See #6199 and [this example](https://docs.flet.dev/services/filepicker/#pick-text-content-and-savedownload-it).
